### ISSUE TITLE
Not able to set 0% for aws_sns_topic using sqs_success_feedback_sample_rate while creating SNS Topic

### ISSUE THREAD

**Issue by prabaharang** [terraform-providers/terraform-provider-aws#11773]
While creating a new SNS Topic resource, the [sqs_success_feedback_sample_rate](https://www.terraform.io/docs/providers/aws/r/sns_topic.html#sqs_success_feedback_sample_rate) is not set as 0%. The value of `0` is set as `NaN`.

If we set any other value other than zero, then the provided value is set correctly. This behavior is happening when creating the Topic. If we update `sqs_success_feedback_sample_rate` (from 5% to 0%) then `0` is set correctly.

### Terraform Version
-  Terraform v0.12.16
- + provider.aws v2.41.0

### Affected Resource(s)
* aws_sns_topic

### Terraform Configuration Files
```hcl
resource "aws_sns_topic" "topic" {
  name                             = topic_name
  sqs_success_feedback_sample_rate = "0"
  sqs_success_feedback_role_arn    = sns_delivery_success_feedback_role.arn
  sqs_failure_feedback_role_arn    = sns_delivery_failure_feedback_role.arn
}
```

### Expected Behavior
sqs_success_feedback_sample_rate should be set correctly.
_Success sample rate_ should be set as 0 or whatever the value (between 0 and 100) being passed in the configuration.

### Actual Behavior
It appears `0` success sample rate is not able to set.

![image](https://user-images.githubusercontent.com/28652963/73190250-b9931c00-414b-11ea-8750-90aa5ea97e9d.png)
In `terraform plan` it shows the correct value but after `terraform apply` it sets as `NaN`.

### Steps to Reproduce
1. Create an SNS Topic with SNS Delivery logging. Set sqs_success_feedback_sample_rate as `0`
2. `terraform plan/apply`
3. View the created SNS Topic in the console and in the `Delivery Status Logging` section, the _Success sample rate_ is showing as `NaN`.

### References
- https://www.terraform.io/docs/providers/aws/r/sns_topic.html
- https://registry.terraform.io/modules/terraform-aws-modules/sns/aws/2.1.0
- https://github.com/terraform-aws-modules/terraform-aws-sns/tree/v2.1.0

**Comment by dreamorosi** [comment]
Have you tried putting the `0` as a number instead of a string?

**Comment by hovmand** [comment]
> Have you tried putting the `0` as a number instead of a string?

I've tried that. Same result.

**Comment by dreamorosi** [comment]
You're right, I've just tried [checked the implementation](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_sns_topic.go#L116) and can confirm that it needs a number.

I have noticed that this happens only when creating a new `aws_sns_topic` resource. Modifying and existing one always results in a correct value on the console, both going from 0 -> 2 and 2 -> 0.
Quite strange because the plan output seems okay:
```log
+ resource "aws_sns_topic" "topic2" {
      + arn                              = (known after apply)
      + id                               = (known after apply)
      + name                             = "github-test-topic-2"
      + policy                           = (known after apply)
      + sqs_failure_feedback_role_arn    = "arn:aws:iam::ACCOUNT_ID:role/sns_delivery_role"
      + sqs_success_feedback_role_arn    = "arn:aws:iam::ACCOUNT_ID:role/sns_delivery_role"
      + sqs_success_feedback_sample_rate = 0
    }
```

**Comment by noam-alchemy** [comment]
Still seeing this as well.

**Comment by husseinmimiHarri** [comment]
any updates ??
